### PR TITLE
TAC updates for new ERP system changes and misc bug fix

### PR DESCRIPTION
**Pull request by spamhurts**
- Added new "template-based" columns to the ehr.project table to accommodate speedkeys. The cost_account data in the account column was moved to the new cost_account column, and the account_id data is now placed in the account field. Speedkey is in the speedkey column.
- updated the source view to supply the new column data
- added default view for ehr.project table
- added the ehr.project table to browse all datasets
- updated the query metadata in demographicsParentStatus.query.xml to fix Issue 339: Incorrect links on parents

### CODE REVIEW

**Review by jallentxbiomed — APPROVED** [review]

File: snprc_ehr/resources/queries/study/BrowseDataSets.sql [hunk 1]
```diff
@@ -68,3 +68,5 @@ UNION
 select 'snprc_ehr' as schema, 'ClinPath' as CategoryId,'Urinalysis Results' as Label, 'HL7UrinalysisPivot' as Name,  true as ShowByDefault, true as isAnimal
 UNION
 select 'ehr' as schema, 'Colony Management' as CategoryId,'Kinship' as Label, 'kinship' as Name,  true as ShowByDefault, true as isAnimal
+UNION
+select 'ehr' as schema, 'Colony Management' as CategoryId,'Project' as Label, 'project' as Name,  true as ShowByDefault, false as isAnimal
```
Comment (RameshRapa) [inline snprc_ehr/resources/queries/study/BrowseDataSets.sql:72]: Is this change on purpose Terry? 'Project' as Label, 'project' as Name, instead of 'Kinship' as Label, 'kinship' as Name,

File: snprc_ehr/resources/queries/study/demographicsParentStatus.query.xml [hunk 2]
```diff
@@ -9,11 +9,7 @@
                     </column>
                     <column columnName="Parent_id">
                         <columnTitle>Parent Id</columnTitle>
-                        <fk>
-                            <fkDbSchema>study</fkDbSchema>
-                            <fkTable>animal</fkTable>
-                            <fkColumnName>id</fkColumnName>
-                        </fk>
+                        <url>ehr/participantView.view?participantId=${Parent_id}</url>
```
Comment (RameshRapa) [inline snprc_ehr/resources/queries/study/demographicsParentStatus.query.xml:12]: what if the Parent_id is not available or null Terry?

Comment (spamhurts) [reply snprc_ehr/resources/queries/study/demographicsParentStatus.query.xml:12]: Then there is no link to click on

File: snprc_ehr/resources/queries/study/BrowseDataSets.sql (hunk 1 above)
Comment (spamhurts) [reply snprc_ehr/resources/queries/study/BrowseDataSets.sql:72]: the table name is ehr.project and the label for it is project

File: snprc_ehr/resources/source_queries/create_v_charge_account.sql [hunk 3]
```diff
@@ -46,6 +48,7 @@ ALTER VIEW [labkey_etl].[v_charge_account] AS
   FROM dbo.charge_account AS ca
     LEFT OUTER JOIN dbo.TAC_COLUMNS AS tc ON tc.object_id = ca.object_id
 	LEFT OUTER JOIN dbo.valid_charge_by_species AS vcs ON ca.charge_id = vcs.charge_id
+    LEFT OUTER JOIN dbo.prd_cost_account AS pca on pca.account_id = ca.account_id
```
Comment (RameshRapa) [inline snprc_ehr/resources/source_queries/create_v_charge_account.sql:51]: I think LEFT OUTER JOIN is same as OUTER JOIN?

Comment (spamhurts) [reply snprc_ehr/resources/source_queries/create_v_charge_account.sql:51]: You would be wrong 🙂🙂

Comment (spamhurts) [reply snprc_ehr/resources/source_queries/create_v_charge_account.sql:51]: https://learn.microsoft.com/en-us/sql/relational-databases/performance/joins?view=sql-server-ver16

- left outer join
- right outer join
- full outer join
-

Comment (spamhurts) [reply snprc_ehr/resources/source_queries/create_v_charge_account.sql:51]: You're probably thinking of a "left join" being the same as a "left outer join"

File: snprc_ehr/resources/queries/study/BrowseDataSets.sql (hunk 1 above)
Comment (spamhurts) [reply snprc_ehr/resources/queries/study/BrowseDataSets.sql:72]: This wasn't a change. It was an additional table

File: snprc_ehr/resources/source_queries/create_v_charge_account.sql (hunk 3 above)
Comment (RameshRapa) [reply snprc_ehr/resources/source_queries/create_v_charge_account.sql:51]: Yes Terry. Correct. LEFT JOIN being same as LEFT OUTER JOIN and we don't usually use 'OUTER' when the same can be done using LEFT JOIN.

**Review by RameshRapa — APPROVED** [review]

File: snprc_ehr/resources/source_queries/create_v_charge_account.sql (hunk 3 above)
Comment (spamhurts) [reply snprc_ehr/resources/source_queries/create_v_charge_account.sql:51]: I always use outer, and I always use inner. But, that's me - I'm an old bugger.
